image/qcow2/qcow2.go

```diff
@@ -752,6 +754,19 @@ func (img *Qcow2) readAtAligned(p []byte, off int64) (int, error) {
 	return n, err
 }
 
+func (img *Qcow2) getL2Table(l1Entry l1TableEntry) ([]l2TableEntry, error) {
+	var err error
+	l2Table := img.l2TablesCache[l1Entry]
+	if l2Table == nil {
+		l2Table, err = readL2Table(img.ra, l1Entry.l2Offset(), img.clusterSize)
+		if err != nil {
+			return nil, err
+		}
+		img.l2TablesCache[l1Entry] = l2Table
```

Comment:
Needs LRU or something to avoid inflating the cache indefinitely

Reply:
For lima use case it is not needed - we import small images. But it LRU makes sense.

Reply:
Right, but this package is expected to be reusable for other projects too.

Reply:
> I like avoiding dependencies and having minimal code that does only what we need and is easy to modify if needed.

👍 
I wish the Go stdlib had LRU...

Reply:
Add the simplest possible lru cache, but kept it as separate package and generic, so we can use it later for caching extended l2 tables or other image formats. The interface is similar to groupcahce/lru so it should be trivial to replace this with another library later.